internal/expressions/binary.go

```diff
@@ -25,7 +25,7 @@ func (b BinaryExpr) TType() ExprType {
 func (b BinaryExpr) ReturnType() ValueType {
 	switch b.Operator.Type {
 	case tokens.Minus, tokens.Star, tokens.Slash, tokens.Percent, tokens.EqualEqual, tokens.BangEqual, tokens.Greater, tokens.GreaterEqual, tokens.Less, tokens.LessEqual:
-		return NumberType
+		return IntType
```

Comment:
Should be updated to support both number types

Reply:
Changed to return an error (ErrorType) so that the compiler can handle the error